dspace-spring-rest/src/main/java/org/dspace/app/rest/repository/CollectionRestRepository.java

```diff
@@ -139,4 +160,104 @@ public CollectionResource wrapResource(CollectionRest collection, String... rels
         return new CollectionResource(collection, utils, rels);
     }
 
+    @Override
+    @PreAuthorize("hasAuthority('ADMIN')")
+    protected CollectionRest createAndReturn(Context context) throws AuthorizeException {
+        HttpServletRequest req = getRequestService().getCurrentRequest().getHttpServletRequest();
+        ObjectMapper mapper = new ObjectMapper();
+        CollectionRest collectionRest;
+        try {
+            ServletInputStream input = req.getInputStream();
+            collectionRest = mapper.readValue(input, CollectionRest.class);
+        } catch (IOException e1) {
+            throw new UnprocessableEntityException("Error parsing request body: " + e1.toString());
+        }
+
+        Collection collection;
+
+
+        try {
+            Community parent = null;
+            String parentCommunityString = req.getParameter("parent");
+            if (StringUtils.isNotBlank(parentCommunityString)) {
+
+                UUID parentCommunityUuid = UUIDUtils.fromString(parentCommunityString);
+                if (parentCommunityUuid == null) {
+                    throw new BadRequestException("The given parent was invalid: "
+                            + parentCommunityString);
+                }
+
+                parent = communityService.find(context, parentCommunityUuid);
+                if (parent == null) {
+                    throw new UnprocessableEntityException("Parent community for id: "
+                            + parentCommunityUuid + " not found");
+                }
+            }
+            collection = cs.create(context, parent);
```

Comment:
I just noticed that `parent` **may be null** at this point.  Shouldn't most/all of the code in this method be surrounded by the `if (StringUtils.isNotBlank(parentCommunityString))`?

Reply:
On second look, it looks like a `null` parent will result in an IllegalArgumentException here: https://github.com/DSpace/DSpace/blob/master/dspace-api/src/main/java/org/dspace/content/CollectionServiceImpl.java#L105

So, we are already protected against a NullPointerException (which was my main worry).  This code may be good enough for now, since the null is handled in the CollectionService.

Reply:
With the checks that happen above the create, namely the checks on the parentCommunityString from the request, the attempt to parse it into a UUID and then the lookup in the DB for said UUID, will already prevent us from ever getting a null parent to the cs.create(context, parent) function. If we were to enter a String in the request parameters that isn't a proper UUID, we'll throw a BadRequestException. If we do enter a proper UUID into the request parameters, but this UUID isn't found as a community, we'll throw an UnprocessableEntity exception.

Reply:
@tdonohue I think something weird is going on in Github which lead to this confusion. The code above this pin isn't updated it seems.. If we go to this link: https://github.com/DSpace/DSpace/pull/2277/files#diff-3cd0c668b5883672718edc7b9da5c7beR195 which is the files changed by this PR, there is an else attached to the if that you mentioned. Indeed with the code that this pin shows, you are correct, but it seems that this is not the latest up to date code for the PR (I added this check in a commit which happened after this discussion started)

Reply:
@Raf-atmire : oh, you are correct!  Sorry about this confusion. It seems I was looking at outdated code (which GitHub was sending me to).  You already added in the `else`, so I'll close out this request. The PR looks great to me -- and I'll update that in my approval below.